frontend/mock-db/init.ts

```diff
@@ -1,57 +1,71 @@
 import { createPerson, getPersonByEmail, type Person } from 'wildebeest/backend/src/activitypub/actors'
-import * as statusesAPI from 'wildebeest/functions/api/v1/statuses'
-import * as reblogAPI from 'wildebeest/functions/api/v1/statuses/[id]/reblog'
-import { statuses } from 'wildebeest/frontend/src/dummyData'
+import { replies, statuses } from 'wildebeest/frontend/src/dummyData'
 import type { Account, MastodonStatus } from 'wildebeest/frontend/src/types'
-
-const kek = 'test-kek'
-/* eslint-disable @typescript-eslint/no-empty-function */
-const queue: unknown = {
-	send() {},
-	sendBatch() {},
-}
-/* eslint-disable @typescript-eslint/no-empty-function */
-const cache: unknown = {
-	get() {},
-	put() {},
-}
+import { createPublicNote, Note } from 'wildebeest/backend/src/activitypub/objects/note'
+import { addObjectInOutbox } from 'wildebeest/backend/src/activitypub/actors/outbox'
+import { insertReblog } from 'wildebeest/backend/src/mastodon/reblog'
+import { insertReply } from 'wildebeest/backend/src/mastodon/reply'
 
 /**
  * Run helper commands to initialize the database with actors, statuses, etc.
  */
 export async function init(domain: string, db: D1Database) {
-	const loadedStatuses: MastodonStatus[] = []
+	const loadedStatuses: { status: MastodonStatus; note: Note }[] = []
 	for (const status of statuses) {
 		const actor = await getOrCreatePerson(domain, db, status.account)
-		loadedStatuses.push(await createStatus(db, actor, status.content))
+		const note = await createStatus(domain, db, actor, status.content)
+		loadedStatuses.push({ status, note })
 	}
 
-	// Grab the account from an arbitrary status to use as the reblogger
-	const rebloggerAccount = loadedStatuses[1].account
-	const reblogger = await getOrCreatePerson(domain, db, rebloggerAccount)
-	// Reblog an arbitrary status with this reblogger
-	const statusToReblog = loadedStatuses[2]
-	await reblogStatus(db, reblogger, statusToReblog, domain)
+	const { reblogger, noteToReblog } = await pickReblogDetails(loadedStatuses, domain, db)
+	reblogNote(db, reblogger, noteToReblog)
+
+	for (const reply of replies) {
+		await createReply(domain, db, reply, loadedStatuses)
+	}
 }
 
 /**
- * Create a status object in the given actors outbox.
+ * Create a status object in the given actor's outbox.
  */
-async function createStatus(db: D1Database, actor: Person, status: string, visibility = 'public') {
-	const body = {
-		status,
-		visibility,
+async function createStatus(domain: string, db: D1Database, actor: Person, content: string) {
+	const note = await createPublicNote(domain, db, content, actor)
+	await addObjectInOutbox(db, actor, note)
+	return note
+}
+
+/**
+ * Reblogs a note (representing a status)
+ */
+async function reblogNote(db: D1Database, reblogger: Person, noteToReblog: Note) {
+	await addObjectInOutbox(db, reblogger, noteToReblog)
+	await insertReblog(db, reblogger, noteToReblog)
+}
```

Comment:
I like the idea of avoiding the full external REST API here, especially given that we don't care about triggering notifications.

Could we move this to somewhere in the `backend/src/` project and reuse it here and in `functions/api/v1/statuses/[id]/reblog.ts`?

And similarly with other functions that are making assumptions about how the internals of the request handlers work?

Reply:
ah yeah that's a good point 😃👍 

sure, I'll merge this if ok and then do some refactoring as a followup 🙂

Reply:
PR opened with the refactoring 🙂  https://github.com/cloudflare/wildebeest/pull/151